### PR TITLE
TE5-T422 Simple ViewPager,ViewPagerTwo and ViePagerWithRecyclerView

### PR DESCRIPTION
**Pull request by vishal2001prajapati**

### CODE REVIEW

File: app/src/main/java/com/example/myapplication3/Fragments/CallFragment.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class CallFragment : Fragment() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View? {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Fragments/CallFragment.kt]: code formatting

File: app/src/main/java/com/example/myapplication3/Fragments/CameraFragment.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.example.myapplication3.R
+
+class CameraFragment : Fragment() {
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
+    ): View? {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Fragments/CameraFragment.kt]: code formatting

File: app/src/main/java/com/example/myapplication3/ViewPager/ViewPageAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            else -> {
+                CameraFragment()
+            }
+        }
+    }
+
+    override fun getPageTitle(position: Int): CharSequence? {
+        return when (position) {
+            ZERO -> {
+                "Camera"
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/ViewPager/ViewPageAdapter.kt]: static string

File: app/src/main/java/com/example/myapplication3/ViewPager/ViewPagerActivity.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_view_pager)
+        supportActionBar?.let {
+            title = getString(R.string.ViewPager)
+            it.setDisplayHomeAsUpEnabled(true)
+        }
+        viewpager = findViewById(R.id.viewPager)
+        tabLayout = findViewById(R.id.tabLayout)
+        viewpager.apply {
+            adapter = ViewPageAdapter(supportFragmentManager)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/ViewPager/ViewPagerActivity.kt]: no need to use apply here for single properties

File: app/src/main/java/com/example/myapplication3/ViewPagerTwo/ViewPagerTwoActivity.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_view_pager_two)
+        viewpagertwo = findViewById(R.id.viewpager_two)
+        tabLayout = findViewById(R.id.viewpagertwo_tablayout)
+        viewpagertwo.adapter = ViewPagerTwoAdapter(this.supportFragmentManager, lifecycle)
+        TabLayoutMediator(tabLayout, viewpagertwo) { tab, position ->
+            when (position) {
+                ZERO -> {
+                    tab.text = "Call"
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/ViewPagerTwo/ViewPagerTwoActivity.kt]: static string

File: app/src/main/java/com/example/myapplication3/ViewPagerTwo/ViewPagerTwoAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    override fun createFragment(position: Int): Fragment {
+        return when (position) {
+            ZERO -> {
+                CameraFragment()
+            }
+            ONE -> {
+                ChatFragment()
+            }
+            TWo -> {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/ViewPagerTwo/ViewPagerTwoAdapter.kt]: spell mistake `TWO`

File: app/src/main/java/com/example/myapplication3/ViewPagerWithRecycler/ViewPagerRecyclerAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        return view == `object`
+    }
+
+    override fun instantiateItem(container: ViewGroup, position: Int): Any {
+        val view = LayoutInflater.from(context).inflate(R.layout.item_viewpager, container, false)
+        val textView: TextView = view.findViewById(R.id.text_viewpager)
+        val imageView: ImageView = view.findViewById(R.id.img_viewpager)
+        val model = modelArrayList[position]
+        val titel = model.str
+        val image = model.img
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/ViewPagerWithRecycler/ViewPagerRecyclerAdapter.kt]: no need to take above three var. You can use directly

File: app/src/main/java/com/example/myapplication3/ViewPagerWithRecycler/ViewPagerWithRecyclerActivity.kt [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        myModelList.apply {
+            add(ViewPagerModel("Antim", R.drawable.antim))
+            add(ViewPagerModel("Keshri", R.drawable.keshri))
+            add(ViewPagerModel("K.G.F", R.drawable.kgf))
+            add(ViewPagerModel("3 idiots", R.drawable.idoits))
+            add(ViewPagerModel("Shershah", R.drawable.shershah))
+        }
+
+        val adapter = ViewPagerRecyclerAdapter(this, myModelList)
+        viewPager.adapter = adapter
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/ViewPagerWithRecycler/ViewPagerWithRecyclerActivity.kt]: use apply

File: app/src/main/res/layout/activity_view_pager_two.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ViewPagerTwo.ViewPagerTwoActivity">
+
+    <com.google.android.material.tabs.TabLayout
+        android:id="@+id/viewpagertwo_tablayout"
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_view_pager_two.xml:10]: make same case id, camel case or underscore

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/myapplication3/Fragments/CallFragment.kt [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.example.myapplication3.R
+
+class CallFragment : Fragment() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Fragments/CallFragment.kt]: empty block

File: app/src/main/java/com/example/myapplication3/ViewPagerWithRecycler/ViewPagerWithRecyclerActivity.kt [hunk 11]
```diff
@@ -0,0 +1,39 @@
+package com.example.myapplication3.ViewPagerWithRecycler
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.viewpager.widget.ViewPager
+import com.example.myapplication3.HUNDRED
+//import com.example.myapplication3.ListView.MyAdapter
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/ViewPagerWithRecycler/ViewPagerWithRecyclerActivity.kt]: commented import

File: app/src/main/res/layout/fragment_camera.xml [hunk 12]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".Fragments.CameraFragment">
+
+    <androidx.appcompat.widget.AppCompatTextView
+        android:id="@+id/camera_fragment"
```
Comment (purvesh-d) [inline app/src/main/res/layout/fragment_camera.xml:9]: You are using underscore pattern or camelcase?

File: app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -67,4 +67,9 @@
     <string name="gridlayout">Gridlayout</string>
     <string name="relativelayout">Relativelayout</string>
     <string name="Grdiwithrecyclerview">Grid Using Recycler</string>
+    <string name="ViewPager">ViewPager</string>
```
Comment (purvesh-d) [inline app/src/main/res/values/strings.xml]: first letter should be small

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/fragment_camera.xml (hunk 12 above)
Comment (vishal2001prajapati) [reply app/src/main/res/layout/fragment_camera.xml:9]: undersscore

**Review by purvesh-d — APPROVED** [review]

**Pull request merged**